fs/smb/client/smb2pdu.c

```diff
@@ -3772,6 +3772,20 @@ int SMB2_query_info(const unsigned int xid, struct cifs_tcon *tcon,
 			  NULL);
 }
 
+#ifdef CONFIG_TRUENAS
+int SMB2_query_streams(const unsigned int xid, struct cifs_tcon *tcon,
+	u64 persistent_fid, u64 volatile_fid, struct smb2_file_stream_info **data,
+	u32 *plen)
+{
+	*plen = 0;
+	return query_info(xid, tcon, persistent_fid, volatile_fid,
+			  FILE_STREAM_INFORMATION, SMB2_O_INFO_FILE, 0,
+			  (sizeof(struct smb2_file_stream_info) + 255) * 32,
+			  sizeof(struct smb2_file_stream_info), (void **)data,
+			  plen);
```

Comment:
> Stream names may be no more than 255 characters in length

Is that 255 UTF-16 "characters" or bytes? SMB2_query_streams() uses some 255 constant, which seems like in bytes. Other code parts in SMB clients I see multiply the length by 2 before writing to StreamNameLength, so odd 255 value there looks odd. :slightly_smiling_face:  Also I see your own comment:

 * NOTE regarding aligment:
 * When multiple FILE_STREAM_INFORMATION data elements are present in the
 * buffer, each MUST be alinged on an 8-byte boundaries; any bytes inserted
 * for alignment SHOULD be set to zero and the receiver MUST ignore them.

Does it mean it should at least be 256 there, or may be even 512 or more?